### PR TITLE
use path.join instead of filepath.join for ssh copy

### PR DESCRIPTION
**Pull request by aaron-prindle**

### CODE REVIEW

File: pkg/provision/buildroot.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func setRemoteAuthOptions(p provision.Provisioner) auth.Options {
 	dockerDir := p.GetDockerOptionsDir()
 	authOptions := p.GetAuthOptions()
 
-	// due to windows clients, we cannot use filepath.Join as the paths
+	// due to windows clients, we cannot use path.Join as the paths
```
Comment (dlorenc) [inline pkg/provision/buildroot.go]: I think you want this comment to stay the same

File: pkg/provision/buildroot.go [hunk 2]
```diff
@@ -170,7 +170,7 @@ func setRemoteAuthOptions(p provision.Provisioner) auth.Options {
 	dockerDir := p.GetDockerOptionsDir()
 	authOptions := p.GetAuthOptions()
```
Comment (dlorenc) [inline pkg/provision/buildroot.go]: Would be good to figure out if we can catch this in a unit test somehow.

File: pkg/provision/buildroot.go (hunk 1 above)
Comment (aaron-prindle) [reply pkg/provision/buildroot.go]: done

**Comment by aaron-prindle** [comment]
This is working on windows with the changes made

**Comment by aaron-prindle** [comment]
I changed the ssh_runner to use path. but kept the exec_runner filepath. as it copies files on the host

**Comment by kfaseela** [comment]
I had installed minikube v0.22.1 from https://github.com/kubernetes/minikube/releases
Are there any updated exe s available with this, or do I have to pull and build?

**Comment by r2d4** [comment]
@kfaseela you can use https://storage.googleapis.com/minikube-builds/1959/minikube-windows-amd64.exe but a new version will be released soon #1971

**Comment by richardjharding** [comment]
I just tried that version above but I still get the following (note the missing ip reported in #1945 seems to be fixed)

```
 Retrying.
E0918 09:13:20.058596   12740 start.go:149] Error starting host:  Error creating host: Error executing step: Provision
g VM.
: Temporary Error: ssh command error:
command : sudo systemctl -f restart docker
err     : Process exited with status 1
output  : Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.

Temporary Error: ssh command error:
command : sudo systemctl -f restart docker
err     : Process exited with status 1
output  : Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.

Temporary Error: ssh command error:
command : sudo systemctl -f restart docker
err     : Process exited with status 1
output  : Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.

Temporary Error: ssh command error:
command : sudo systemctl -f restart docker
err     : Process exited with status 1
output  : Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.

Temporary Error: ssh command error:
command : sudo systemctl -f restart docker
err     : Process exited with status 1
output  : Job for docker.service failed because the control process exited with error code.
See "systemctl status docker.service" and "journalctl -xe" for details.
```

**Comment by sarraRoza** [comment]
Hello @richardjharding I had the same problem like yours when I was using Hyper-v because my VM wasn't correctly connected to Internet ! Moved to VirtualBox and the problem was solved!

@r2d4  the fixed version works for me but I'm getting a new error.

```
C:\WINDOWS\system32>minikube start
Starting local Kubernetes v1.7.5 cluster...
Starting VM...
Getting VM IP address...
Moving files into cluster...
Setting up certs...
Connecting to cluster...
Setting up kubeconfig...
Starting cluster components...
Kubectl is now configured to use the cluster.

C:\WINDOWS\system32>minikube status
minikube: Running
cluster: Running
kubectl: Correctly Configured: pointing to minikube-vm at 192.168.99.100
```

None of kubectl commands works for me, even `minikube dashboard` gives connection refused error.

```
C:\WINDOWS\system32>kubectl get pods --all-namespaces
Unable to connect to the server: dial tcp 192.168.99.100:8443: connectex: Aucune connexion n’a pu être établie car l’ordinateur cible l’a expressément refusée.
```

**Comment by sbueringer** [comment]
Seems like the certs are setup correctly now:
````
ls -la /var/lib/localkube/certs
total 40
drwxr-xr-x 2 root root 4096 Sep 23 16:27 .
drwxr-xr-x 4 root root 4096 Sep 23 16:27 ..
-rw-r--r-- 1 root root 1269 Sep 23 16:27 apiserver.crt
-rw------- 1 root root 1675 Sep 23 16:27 apiserver.key
-rw-r--r-- 1 root root 1066 Sep 23 16:27 ca.crt
-rw------- 1 root root 1679 Sep 23 16:27 ca.key
-rw-r--r-- 1 root root 1074 Sep 23 16:27 proxy-client-ca.crt
-rw------- 1 root root 1671 Sep 23 16:27 proxy-client-ca.key
-rw-r--r-- 1 root root 1103 Sep 23 16:27 proxy-client.crt
-rw------- 1 root root 1679 Sep 23 16:27 proxy-client.key
````

 I get errors on startup of localkube. Localkube seems to search at the wrong place for the certificates:
````
journalctl -f -u localkube
 I0923 16:39:43.699081    6102 ready.go:32] Performing healthcheck on https://localhost:8443/healthz
Sep 23 16:39:43 gssnplus localkube[6102]: I0923 16:39:43.708378    6102 ready.go:67] Got healthcheck response: ok
Sep 23 16:39:43 gssnplus localkube[6102]: apiserver is ready!
Sep 23 16:39:43 gssnplus localkube[6102]: Starting controller-manager...
Sep 23 16:39:43 gssnplus localkube[6102]: Waiting for controller-manager to be healthy...
Sep 23 16:39:43 gssnplus localkube[6102]: I0923 16:39:43.708443    6102 controllermanager.go:108] Version: v1.7.5
Sep 23 16:39:43 gssnplus localkube[6102]: controller-manager: Exit with error: invalid configuration: [unable to read client-cert /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt: no such file or directory, unable to read client-key /var/lib/localkube/\var\lib\localkube\certs\apiserver.key for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.key: no such file or directory, unable to read certificate-authority /var/lib/localkube/\var\lib\localkube\certs\ca.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\ca.crt: no such file or directory]
Sep 23 16:39:43 gssnplus localkube[6102]: I0923 16:39:43.910460    6102 controllermanager.go:108] Version: v1.7.5
Sep 23 16:39:43 gssnplus localkube[6102]: E0923 16:39:43.910601    6102 controllermanager.go:116] unable to register configz: register config "componentconfig" twice
Sep 23 16:39:43 gssnplus localkube[6102]: controller-manager: Exit with error: invalid configuration: [unable to read client-cert /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt: no such file or directory, unable to read client-key /var/lib/localkube/\var\lib\localkube\certs\apiserver.key for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.key: no such file or directory, unable to read certificate-authority /var/lib/localkube/\var\lib\localkube\certs\ca.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\ca.crt: no such file or directory]
Sep 23 16:39:44 gssnplus localkube[6102]: I0923 16:39:44.111675    6102 controllermanager.go:108] Version: v1.7.5
Sep 23 16:39:44 gssnplus localkube[6102]: E0923 16:39:44.111941    6102 controllermanager.go:116] unable to register configz: register config "componentconfig" twice
Sep 23 16:39:44 gssnplus localkube[6102]: controller-manager: Exit with error: invalid configuration: [unable to read client-cert /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt: no such file or directory, unable to read client-key /var/lib/localkube/\var\lib\localkube\certs\apiserver.key for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.key: no such file or directory, unable to read certificate-authority /var/lib/localkube/\var\lib\localkube\certs\ca.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\ca.crt: no such file or directory]
Sep 23 16:39:44 gssnplus localkube[6102]: I0923 16:39:44.313584    6102 controllermanager.go:108] Version: v1.7.5
Sep 23 16:39:44 gssnplus localkube[6102]: E0923 16:39:44.313956    6102 controllermanager.go:116] unable to register configz: register config "componentconfig" twice
Sep 23 16:39:44 gssnplus localkube[6102]: controller-manager: Exit with error: invalid configuration: [unable to read client-cert /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt: no such file or directory, unable to read client-key /var/lib/localkube/\var\lib\localkube\certs\apiserver.key for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.key: no such file or directory, unable to read certificate-authority /var/lib/localkube/\var\lib\localkube\certs\ca.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\ca.crt: no such file or directory]
Sep 23 16:39:44 gssnplus localkube[6102]: I0923 16:39:44.515916    6102 controllermanager.go:108] Version: v1.7.5
Sep 23 16:39:44 gssnplus localkube[6102]: E0923 16:39:44.516407    6102 controllermanager.go:116] unable to register configz: register config "componentconfig" twice
Sep 23 16:39:44 gssnplus localkube[6102]: controller-manager: Exit with error: invalid configuration: [unable to read client-cert /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt: no such file or directory, unable to read client-key /var/lib/localkube/\var\lib\localkube\certs\apiserver.key for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.key: no such file or directory, unable to read certificate-authority /var/lib/localkube/\var\lib\localkube\certs\ca.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\ca.crt: no such file or directory]
Sep 23 16:39:44 gssnplus localkube[6102]: controller-manager is ready!
Sep 23 16:39:44 gssnplus localkube[6102]: Starting scheduler...
Sep 23 16:39:44 gssnplus localkube[6102]: Waiting for scheduler to be healthy...
Sep 23 16:39:44 gssnplus localkube[6102]: scheduler: Exit with error: unable to create kube client: unable to build config from flags: invalid configuration: [unable to read client-cert /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.crt: no such file or directory, unable to read client-key /var/lib/localkube/\var\lib\localkube\certs\apiserver.key for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\apiserver.key: no such file or directory, unable to read certificate-authority /var/lib/localkube/\var\lib\localkube\certs\ca.crt for gssnplus due to open /var/lib/localkube/\var\lib\localkube\certs\ca.crt: no such file or directory]
....
````

**Pull request merged**